core/src/debug_client.rs

```
@@ -0,0 +1,65 @@
+use reqwest;
+use prost::Message;
+use temporal_sdk_core_protos::temporal::api::history::v1::History;
+use std::time::Duration;
+
+struct DebugClient {
+    debugger_url: String,
+    _client: reqwest::Client
+}
+
+impl DebugClient {
+
+    fn new(url: String) -> DebugClient {
+        DebugClient {
+            debugger_url: url,
+            _client: reqwest::Client::new(),
+        }
+    }
+
+    async fn get_history(&self) -> Result<History, anyhow::Error> {
+        let url = self.debugger_url.clone() + "/history";
+        let resp = self._client.get(url)
+            .header("Temporal-Client-Name", "temporal-core")
+            .header("Temporal-Client-Version", "0.1.0")
+            .send()
+            .await?;
+
+        let bytes = resp.bytes().await?;
+        Ok(History::decode(bytes)?) // decode_length_delimited() does not work
+
+    }
+
+    async fn post_wft_started(&self, event_id: &i64) -> bool {
+        let url = self.debugger_url.clone() + "/current-wft-started";
+        let resp = self._client.get(url)
+            .header("Temporal-Client-Name", "temporal-debug-core")
+            .header("Temporal-Client-Version", "0.1.0")
+            .timeout(Duration::from_secs(5))
+            .json(event_id)
+            .send()
+            .await;
+
+        match resp {
+            Ok(r) => { r.status() == reqwest::StatusCode::OK },
+            Err(_) => false
+        }
```

This hides the error in the event something went wrong - I think logging a warning here makes sense, and probably also just returning the result to the caller makes sense.